pandas/core/nanops.py

```diff
@@ -367,19 +366,23 @@ def _wrap_results(result, dtype: np.dtype, fill_value=None):
                 result = np.datetime64("NaT", "ns")
             else:
                 result = np.int64(result).view("datetime64[ns]")
+            # retain original unit
+            result = result.astype(dtype, copy=False)
```

Comment:
So do these changes make things internally More Correct and not necessarily have an external behavior change?

Reply:
not user-facing unless they are directly using nanops, which they shouldnt be